src/Illuminate/Foundation/Bootstrap/RegisterProviders.php

```diff
@@ -57,7 +58,7 @@ protected function mergeAdditionalProviders(Application $app)
         $app->make('config')->set(
             'app.providers',
             array_merge(
-                $app->make('config')->get('app.providers'),
+                $app->make('config')->get('app.providers') ?? ServiceProvider::defaultProviders()->toArray(),
```

Comment:
Fixed scenario where `app.php` configuration file is not available. Verified with tests below.